presto-common/src/main/java/com/facebook/presto/common/type/TypeUtils.java

```diff
@@ -35,6 +41,28 @@ private TypeUtils()
     {
     }
 
+    public static boolean isNumericType(Type type)
```

Comment:
Since you are using composition for `isNonDecimalNumericType`, should this be changed to `isNonDecimalNumericType(type) || type instance DecimalType` then?

Reply:
Done